src/coreComponents/physicsSolvers/SolverBase.hpp

```diff
@@ -648,6 +648,44 @@ class SolverBase : public ExecutableGroup
 
   virtual bool registerCallback( void * func, const std::type_info & funcType ) final override;
 
+
+  /**
+   * This function provides capability to post process input values prior to
+   * any other initialization operations.
+   */
+  virtual void postProcessInput() override {}
+
+  /**
+   * @brief Called by Initialize() prior to initializing sub-Groups.
+   */
+  virtual void initializePreSubGroups() override {}
+
+  /**
+   * @brief Called by Initialize() after to initializing sub-Groups.
+   */
+  virtual void initializePostSubGroups() override {}
+
+  /**
+   * @brief Called by InitializePostInitialConditions() prior to initializing sub-Groups.
+   */
+  virtual void initializePostInitialConditionsPreSubGroups() override {}
+
+  /**
+   * @brief Called by InitializePostInitialConditions() after to initializing sub-Groups.
+   */
+  virtual void initializePostInitialConditionsPostSubGroups() override {}
+
+  /**
+   * @brief Performs initialization required after reading from a restart file.
+   */
+  virtual void postRestartInitialization() override {}
+
+  /**
+   * @brief Performs re-initialization of certain variable depending on the solver being used.
+   */
+  virtual void reinit() override {}
```

Comment:
What's the reason to add these?

Reply:
The reason originally was to be able to access all these reinitialization routines from the PySolver to be able to call them from python and then simplify multi-realization. 
We had a discussion with Josh and Randy and it has been decided to create the "reinit" routine in Group.hpp and access it in the PySolver interface, then this routine is overriden by each solver dependeing on the variables to re-initialize and how all this other initialization routines (postProcessInput etc.) are overriden.

Reply:
I guess for example `postProcessInput()` already has that exact definition in `Group.hpp` https://github.com/GEOSX/GEOSX/blob/1ac5032a4544f8d7d7828b10df1a9142ba7496a2/src/coreComponents/dataRepository/Group.hpp#L1300

So what's the point of re-defining it again?

Reply:
The re-definition of these was to make them `Public` for the `Solvers` since they are `Protected` in `Group.hpp`. But since we now use the `reinit()` I just need this one function to be public in the different solvers. I can now remove all of the other re-definition.

Reply:
Ahh that makes sense, thanks!